File: backend/app/schemas/stats.py
from datetime import date

from pydantic import BaseModel


class CalendarDaySummary(BaseModel):
    date: date
    scheduled: int
    completed: int
    percent: int | None

class CalendarSummaryResponse(BaseModel):
    days: list[CalendarDaySummary]

class HabitStatsResponse(BaseModel):
    current_streak: int
    longest_streak: int
    total_completions: int
    weekly_completion_percent: int | None
    monthly_completion_percent: int | None

class AccountStatsResponse(BaseModel):
    total_active_habits: int
    weekly_completion_percent: int | None
    best_current_streak: int
    total_lifetime_completions: int

File: mobile/src/api/stats.ts
import { apiClient } from "./client";

export interface AccountStats {
  total_active_habits: number;
  weekly_completion_percent: number | null;
  best_current_streak: number;
  total_lifetime_completions: number;
}

export interface HabitStats {
  current_streak: number;
  longest_streak: number;
  total_completions: number;
  weekly_completion_percent: number | null;
  monthly_completion_percent: number | null;
}

export async function getStatsOverview(): Promise<AccountStats> {
  const { data } = await apiClient.get<AccountStats>("/stats/overview");
  return data;
}

export async function getHabitStats(habitId: string): Promise<HabitStats> {
  const { data } = await apiClient.get<HabitStats>(`/habits/${habitId}/stats`);
  return data;
}
